src/main/java/it/gov/pagopa/cgn/portal/service/AgreementService.java

```diff
@@ -0,0 +1,71 @@
+package it.gov.pagopa.cgn.portal.service;
+
+import it.gov.pagopa.cgn.portal.enums.AgreementStateEnum;
+import it.gov.pagopa.cgn.portal.exception.InvalidRequestException;
+import it.gov.pagopa.cgn.portal.model.AgreementEntity;
+import it.gov.pagopa.cgn.portal.model.AgreementUserEntity;
+import it.gov.pagopa.cgn.portal.repository.AgreementRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+@Service
+public class AgreementService {
+
+    private final AgreementRepository agreementRepository;
+
+    private final AgreementUserService userService;
+
+    public AgreementEntity findById(String subscriptionId) {
+        return agreementRepository.findById(subscriptionId)
+                .orElseThrow(() -> new InvalidRequestException("Agreement not found"));
```

Comment:
I don't remember if an agreement is not found, should we return 404 ?

Reply:
> I don't remember if an agreement is not found, should we return 404 ?

No, if an agreement doesn't exists will be created empty. All profiles must be have one. 
This method is called by Discount and Profile service to check if the agreement passed by FE exists. In the future version this method should check even if the current user matches with the agreement passed by FE (throw JWT token).
For now, if the agreement passed by FE is invalid (not exists), a 400 will be returned.

Reply:
ok 👍